File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/presenter/CardFilterPresenter.java
package com.dbottillo.mtgsearchfree.presenter;

import com.dbottillo.mtgsearchfree.model.CardFilter;
import com.dbottillo.mtgsearchfree.view.CardFilterView;

public interface CardFilterPresenter {

    void init(CardFilterView view);

    void loadFilter();

    void update(CardFilter.TYPE type, boolean on);

}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/presenter/CardFilterPresenter.kt
package com.dbottillo.mtgsearchfree.presenter

import com.dbottillo.mtgsearchfree.resources.CardFilter
import com.dbottillo.mtgsearchfree.view.CardFilterView

interface CardFilterPresenter {

    fun init(view: CardFilterView)

    fun loadFilter()

    fun update(type: CardFilter.TYPE, on: Boolean)

}